### ISSUE TITLE
[Enhancement] Connect to a running drone at startup

### ISSUE THREAD

**Issue by b-ehlers** [pstavirs/ostinato#322]
I like to start the drone before starting ostinato.

Currently I have two options:

* Without option `-c` ostinato always starts the drone. That results in an error popup, as a drone is already active. Then it connects to the local drone, what succeeds. So it basically works, but the error popup is annoying.
* With option `-c` it doesn't connect to any drone, that has to be done manually.

So both options are not ideal for me. I would like, that it starts a drone only, when an initial connection to the local drone fails. Alternatively a command line option to connect to a new port group (which doesn't need to be a local one) would be fine.

**Comment by anubisg1** [comment]
This seems related to #320

**Comment by pstavirs** [comment]
@b-ehlers  I have the same problem during development but thought no one else would have this problem. So I'm curious to know why do you need or prefer to start drone and ostinato separately?

**Comment by b-ehlers** [comment]
The drone is reliably started, when starting the GUI, so I haven't noticed any bug in this area.

But I'm using Ostinato mainly within GNS3 in a VM and I want to be able to connect to the drone directly after starting it. Therefore I'm starting the drone during VM's boot. For lazy people, that don't want to use an external Ostinato GUI, the VM also contains a GUI. And when that GUI is started I'm getting two error messages, that the GUI can't start the drone (as that is already running).

My current workaround is a wrapper script, that tests, if the drone is already running. Only if the drone is not active, the wrapper script starts the real drone program.

**Comment by pstavirs** [comment]
> I would like, that it starts a drone only, when an initial connection to the local drone fails.

With the current implementation of starting a local _drone_ and then connecting to local portgroup, we already have a delay in the port group getting connected and ports being listed in the GUI because drone takes some time to enumerate all the ports (essentially a libpcap API that takes time). Adding a initial connection check to drone will add further delay to this because if no drone is running, the connect API will retry a couple of times before timing out.

> Alternatively a command line option to connect to a new port group (which doesn't need to be a local one) would be fine.

How would this help? Ostinato will try and start drone. This new option should also do the `-c` functionality?

What if you create a session file with just the local port group and nothing configured on any ports. Then start Ostinato using `-c` and the session file. Would that work?

**Comment by b-ehlers** [comment]
Well, I have a workaround in place, so this issue is no longer relevant. Alternatively, as mentioned by @pstavirs a session file could be used.

Closing it.
